### PR TITLE
Improve the test suite

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

- Remove `full` ENV var
- Add `test` Rake task
- Combine all spec helper files
- Install Codecov

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -14,25 +29,30 @@
   config.filter_sensitive_data('API_TOKEN') { ENV['MASTER_JWT'] }
 end
 
-mode = ENV['MODE'] || 'unit'
```
Comment (joshcanhelp) [inline spec/spec_helper.rb]: Removed the test mode now that we're using VCR

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -14,25 +29,30 @@
...
-
 $LOAD_PATH.unshift File.expand_path('..', __FILE__)
 $LOAD_PATH.unshift File.expand_path('../../lib', __FILE__)
 
-require 'rspec'
```
Comment (joshcanhelp) [inline spec/spec_helper.rb]: Moved up

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -14,25 +29,30 @@
...
+  raise e if elapsed_time >= time
+  sleep increment
+  wait(time, increment, elapsed_time + increment, &block)
+end
+
 def entity_suffix
   'rubytest'
 end
 
-puts "Entity suffix is #{entity_suffix}"
```
Comment (joshcanhelp) [inline spec/spec_helper.rb]: Moving test mode-specific code

File: Gemfile [hunk 4]
```diff
@@ -13,4 +13,5 @@ end
 group :test do
   gem 'webmock', require: false
   gem 'vcr', require: false
+  gem 'codecov', :require => false
```
Comment (lbalmaceda) [inline Gemfile]: why the arrow?

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -6,6 +20,7 @@
 
 require 'vcr'
 VCR.configure do |config|
+  # Uncomment the line below to record new cassettes
```
Comment (lbalmaceda) [inline spec/spec_helper.rb]: what does that mean? should it be clarified?

File: spec/spec_helper.rb (hunk 1 above)
Comment (lbalmaceda) [reply spec/spec_helper.rb]: VCR?

File: spec/spec_helper.rb (hunk 2 above)
Comment (lbalmaceda) [reply spec/spec_helper.rb]: down you mean? :trollface:

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -14,25 +29,30 @@
...
-require 'rspec'
-require 'rack/test'
-require 'faker'
-require 'auth0'
-
 Dir['./lib/**/*.rb'].each { |f| require f }
 Dir['./spec/support/**/*.rb'].each { |f| require f }
 Dir['./spec/support/*.rb'].each { |f| require f }
 
+require 'rspec'
```
Comment (lbalmaceda) [inline spec/spec_helper.rb]: cosmetic but, shouldn't this be on the top of the file instead of flying here?

Comment (joshcanhelp) [reply spec/spec_helper.rb]: Actually no ... `config.include Credentials` needs to happen after the file requires

**Pull request merged**